Source/Core/Core/IOS/Network/KD/Mail/WC24Send.cpp

```diff
@@ -27,6 +30,22 @@ void WC24SendList::ReadSendList()
     return;
   }
 
+  // Make sure that next_entry_offset is not out of bounds.
+  if (m_data.header.next_entry_offset > sizeof(MailListEntry) * MAX_ENTRIES)
```

Comment:
If what you just said about the index/offset relationship, this is not enough. `offset` must be exactly divisible by `128`, nonzero, and not exceed `128+(MAX_ENTRIES-1)*128`.